axlearn/common/state_builder.py

```diff
@@ -912,7 +912,27 @@ def replace_prefix(source_weight, target_weight):
             )(source_weight, target_weight)
         elif cfg.strategy == "keep_target":
             out["pos_emb"] = target["pos_emb"]
-        # TODO(xianzhi): Implement interpolation for shape matching.
+        elif cfg.strategy.startswith("interpolate"):
+
+            def interpolate(x):
+                if cfg.strategy == "interpolate":
+                    return jax.image.resize(
+                        x, (x.shape[0], target_len, x.shape[2]), method="bilinear", antialias=False
+                    )
+                elif cfg.strategy == "interpolate_with_cls_token_removed":
```

Comment:
If the strategy gets more complicated, we should make it a Union[str, InstantiableConfig ].